src/main/java/net/atos/entng/rbs/service/BookingServiceSqlImpl.java

```diff
@@ -1099,26 +1099,21 @@ Future<JsonObject> checkRightsAndDeleteBooking(String bookingStringId, UserInfos
 				})
 				.onSuccess(hasRight -> {
 					if (Boolean.TRUE.equals(hasRight)) {
-						promise.complete(new JsonObject()
-								.put(Field.ID, bookingStringId)
-								.put(Field.STATUS, Field.ERROR)
-								.put(Field.MESSAGE, "Temporary result")
-						);
-//						this.deleteBooking(bookingStringId, user)
-//								.onSuccess(res -> {
-//									promise.complete(new JsonObject()
-//											.put(Field.ID, bookingStringId)
-//											.put(Field.STATUS, Field.OK)
-//											.put(Field.MESSAGE, "Successfully deleted")
-//									);
-//								})
-//								.onFailure(err -> {
-//									promise.complete(new JsonObject()
-//											.put(Field.ID, bookingStringId)
-//											.put(Field.STATUS, Field.ERROR)
-//											.put(Field.MESSAGE, err.getMessage())
-//									);
-//								});
+						this.deleteBooking(bookingStringId, user)
+								.onSuccess(res -> {
+									promise.complete(new JsonObject()
+											.put(Field.ID, bookingStringId)
+											.put(Field.STATUS, Field.OK)
+											.put(Field.MESSAGE, "Successfully deleted")
+									);
+								})
+								.onFailure(err -> {
+									promise.complete(new JsonObject()
```

Comment:
tu complete quoi qu'il en soit ? c'est un peu contre intuitif non ?

Reply:
/**
	 * This methods checks if the user has the right to delete the booking and if so deletes it
	 * All cases lead to completed promises in order for checkRightsAndDeleteBookings() to gather all the booking deletion info
	 * and send it back to Calendar Module using event bus response.
	 * Example of completed promise :
	 * {
	 *     "id": "000",
	 *     "status": "error",
	 *     "message": "Unauthorized"
	 * }
	 * @param bookingStringId {@link String} the id of the booking
	 * @param user {@link UserInfos} the user
	 * @return {@link Future<JsonObject>} (see example above)
	 */